octodns/provider/constellix.py

```diff
@@ -161,9 +162,52 @@ def record_delete(self, zone_name, record_type, record_id):
             record_type = 'ANAME'
 
         zone_id = self.domains.get(zone_name, False)
-        path = '/{}/records/{}/{}'.format(zone_id, record_type, record_id)
+        path = '/domains/{}/records/{}/{}'.format(zone_id, record_type,
+                                                  record_id)
         self._request('DELETE', path)
 
+    def pools(self, pool_type):
+        if self._pools is None:
```

Comment:
It looks like the parameter `pool_type` will make a difference in the API call made, but this will cache things the first time through and then just short-circuit and return the cached value from then on. Seems like if this was called with an `A` and then later with `AAAA` it'd return the `A`s the second time.

Reply:
I've updated this logic to now check the pool type as well as the name. I've also added unit tests to verify this behaviour.